crossplane-contrib/provider-gcp
# Enable ESS plugin

### Description of your changes
This PR consumes ESS plugin changes in crossplane-runtime and loads and passes TLS config to the controllers. 

Fixes #

I have:


### How has this code been tested
E2E test with following steps:

1) Create a local kind cluster
2) Prepare Vault as described [here](https://github.com/crossplane/docs/blob/master/content/knowledge-base/integrations/vault-as-secret-store.md#prepare-vault)
3) Build and install Crossplane on master
```
helm upgrade --install crossplane crossplane-master/crossplane --namespace crossplane-system --create-namespace --set args={--debug,--enable-external-secret-stores} 
```
4) Build and install provider from this branch
```
kubectl apply -f provider-gcp-ess.yaml
```
5) Install storeconfigs
```
kubectl apply -f storeconfig-xp.yaml -f storeconfig-gcp.yaml
```
6) Install [ESS plugin Vault](https://github.com/crossplane-contrib/ess-plugin-vault)
```
helm upgrade --install ess-plugin-vault oci://xpkg.upbound.io/crossplane-contrib/ess-plugin-vault --namespace crossplane-system -f values.yaml
```
7) Install vaultconfig
```
kubectl apply -f vaultconfig.yaml
```
8) Install XRD and composition:
```
kubectl apply -f xrd.yaml -f composition.yaml
```
9) Install claim
```
kubectl apply -f claim.yaml
```
10) Portforward Vault service and verify that secrets are written
```
kubectl port-forward svc/vault 8200:8200 -n vault-system
```
11) Delete claim and verify that secrets are deleted from Vault
```
kubectl delete -f claim.yaml
```

All the files mentioned above can be found [here](https://gist.github.com/ezgidemirel/190721cbc6e3b3449697888dbc001911).



[contribution process]: https://git.io/fj2m9
